pkg/dss/auth/auth.go

```diff
@@ -140,19 +139,18 @@ func (r *JWKSResolver) ResolveKey(ctx context.Context) (interface{}, error) {
 
 // Authorizer authorizes incoming requests.
 type Authorizer struct {
-	logger           *zap.Logger
-	keyGuard         sync.RWMutex
-	key              interface{}
-	requiredScopes   map[string][]string
-	requiredAudience string
+	logger            *zap.Logger
+	key               interface{}
+	requiredScopes    map[string][]string
+	requiredAudiences map[string]bool
```

Comment:
nit: `map[string]struct{}` is typically used for set-like semantics iirc.

Reply:
ehh I agree, but tbh I'm not a huge fan of those semantics unless we really need to squeeze out some performance gains. In reality people won't be specifying any huge number of maps, and I think this is both easier to read and easier to use ie: 

if aud["key"] {...}

 vs

 if _, ok := aud["key"]; ok {...}

Reply:
FWIW, [the Go blog](https://blog.golang.org/maps) mentions (endorses?) the "`map[ValueType]bool` -> set" pattern

Reply:
(my weak recommendation is to keep as-is (`map[string]bool`), but defer to both of you with better Go experience)